### PR TITLE
Unnest nested spec type definitions

### LINKED ISSUE #59: Remove nested type definition specifications
The type definition of `IntracellularRecordingsTable` includes the type definitions for `IntracellularElectrodesTable`, `IntracellularStimuliTable`, and `IntracellularResponsesTable`. I understand that these three types are meant to be used only within the `IntracellularRecordingsTable`, but these types can still technically be used elsewhere, and if these types are defined outside of `IntracellularRecordingsTable`, it would be easier to parse the schema and avoid edge cases that arise from having nested type definitions. 

I suggest moving the specifications for these types to the root level of the extension and using `neurodata_type_inc` to include them in the `IntracellularElectrodesTable` type.

### PR DESCRIPTION
**Pull request by rly**
Fix #59

Also use `unittest` instead of `unittest2`

### CODE REVIEW

File: spec/ndx-icephys-meta.extensions.yaml [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
   neurodata_type_inc: DynamicTable
-  doc: Container for storing a collection of DynamicTables that are aligned by row
+  doc: DynamicTable container that subports storing a collection of subtables. Each
+    sub-table is a DynamicTable itself that is aligned with the main table by row
     index. I.e., all DynamicTables stored in this group MUST have the same number
-    of rows. This type effectively defines a 2-level table in which columns of the
-    table a grouped into categories, with eachDynamicTable respresenting a particular
-    category.
+    of rows. This type effectively defines a 2-level table in which the main data
+    is stored in the main table implementd by this typeand additional columns of the
```
Comment (oruebel) [inline spec/ndx-icephys-meta.extensions.yaml:8]:
```suggestion
    is stored in the main table implemented by this type and additional columns of the
```

File: spec/ndx-icephys-meta.extensions.yaml [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+    table are grouped into categories, with each category beingrepresented by a separate
```
Comment (oruebel) [inline spec/ndx-icephys-meta.extensions.yaml:9]:
```suggestion
    table are grouped into categories, with each category being represented by a separate
```

File: spec/ndx-icephys-meta.extensions.yaml (hunk 1 above)
Comment (oruebel) [reply spec/ndx-icephys-meta.extensions.yaml:8]: Should be changed in the python script.

File: spec/ndx-icephys-meta.extensions.yaml (hunk 2 above)
Comment (oruebel) [reply spec/ndx-icephys-meta.extensions.yaml:9]: Should be changed in the python script.

**Comment by rly** [comment]
I fixed those typos and others in a new commit.

**Comment by oruebel** [comment]
@rly can you rerun the create_extension_spec.py script. It looks like the typos are fixed in the script but not the schema. Otherwise this looks good.

**Comment by rly** [comment]
My bad. Fixed now. Feel free to merge. Also see #70 which is needed with the latest dev branch of HDMF to get all tests working.

**Pull request merged**